pkg/actuators/machine/reconciler.go

```diff
@@ -175,9 +190,43 @@ func (r *Reconciler) reconcileMachineWithCloudState(conditionFailed *ibmcloudpro
 	// conditionFailed is nil, get the cloud instance and reconcile the fields
 	newInstance, err := r.ibmClient.InstanceGetByName(r.machine.Name, r.providerSpec)
 	if err != nil {
+		// Check whether the machine was recently removed to replace a stuck machine in order to resolve
+		// https://issues.redhat.com/browse/OCPBUGS-1327
+		// We need to wait until the IBM Cloud VSI is actually gone before re-creating it with the same name (to prevent cascading issues using a different name)
+		// Check if the machine just completed deletion and should now have a replacement creation in progress
+		if conditionTypeCheck(r.providerStatus.Conditions, ibmcloudproviderv1.MachineReplaced) != nil {
+			klog.Infof("%s: waiting for replacement machine to start creating", r.machine.Name)
+			return nil
+		}
 		return fmt.Errorf("get instance failed with an error: %q", err)
 	}
 
+	if newInstance.Status != nil && *newInstance.Status == statusDeleting && conditionTypeCheck(r.providerStatus.Conditions, ibmcloudproviderv1.MachineReplaced) != nil {
+		klog.Infof("%s: waiting for stuck machine to be deleted prior to replacement", r.machine.Name)
+		return nil
+	}
+
+	// Check whether the machine remains in Provisioned state but not Running for more than 'instanceReplaceDeadline' minutes
+	// Attempt to mitigate https://issues.redhat.com/browse/OCPBUGS-1327
+	// Bypass check if machine statuses do not match this expected case, and only check for stuck machine if the machine has not yet been replaced (attempt to replace only once)
+	if newInstance.Status != nil && *newInstance.Status == statusRunning && r.machine.Status.Phase != nil && *r.machine.Status.Phase == phaseProvisioned && conditionTypeCheck(r.providerStatus.Conditions, ibmcloudproviderv1.MachineReplaced) == nil {
+		if replacementRequired, err := r.checkInstanceRequiresReplacement(newInstance); err == nil && replacementRequired {
+			klog.Warningf("%s: attempting to replace stuck machine", r.machine.Name)
+			// Update status that machine will be replaced
+			r.providerStatus.Conditions = reconcileProviderConditions(r.providerStatus.Conditions, ibmcloudproviderv1.IBMCloudMachineProviderCondition{
+				Type:    ibmcloudproviderv1.MachineReplaced,
+				Reason:  ibmcloudproviderv1.MachineReplacementRequested,
+				Message: machineReplacementRequestedMessageCondition,
+				Status:  apicorev1.ConditionTrue,
+			})
+			// We must remove the machine's addresses and provider ID since we are replacing it
+			r.machine.Spec.ProviderID = nil
+			r.machine.Status.Addresses = make([]apicorev1.NodeAddress, 0)
```

Comment:
This is not what Machine API is going to expect. The core controller doesn't expect these things to go backwards. I'm not sure how this would behave, but you've effectively gone from `Provisioned` to `Provisioning` which is not expected.

If you only set the `ProviderID`, then it will stay in provisioning, you can then set the addresses once you are confident the Machine will become a Node, is that possible here? I guess we need the addresses to be set for the CSR to be approved?

Is there any way to know the Node has ignited properly without waiting for the Node object to appear? Cold afterburn possibly set a tag or some metadata on the instance (we do this on Azure for example), and then once we observe that, set the node addresses to move into Provisioned

Reply:
I'll try to digest this, when I am able. But I believe if these two items are set, the Machine API will not re-create the underlying IBM VSI, and go into a failed state instead......if I recall correctly.

Will try to investigate into that

Reply:
In those other avenues, can we achieve those changes by Code Freeze (2022/10/28)?

I can try to find the Azure example to get a better idea of impact, if we can get done for Code Freeze

Reply:
I would suggest if you are clearing these to also clear the phase to "reset" the machine back to the beginning

Reply:
I've made some significant changes to incorporate some of these items, to get a more consistent and reliable path.
Hopefully this pathway provides the following:
- A reliable method for determining when we have this "stuck" machine
- Proper data updates to make sure the MAO doesn't immediately flag the machine is failed, during the replacement
- A method that prevents a single API call failure to update the Machine resource does not break the entire replacement strategy
- A wait period for the stuck machine to delete
- A series of status conditions that provide details on the progress of the replacement
- A method to perform the replacement once for a machine, whereas a user would followup with manual mitigation (`oc delete machine`), if necessary, since all attempts to replace have been successful in testing

I hope this pathway is sufficient without adding too much complexity, since we expect this to survive in the 4.12 release, unless RHEL 9 becomes available, and hopefully 4.13 will contain the necessary RHEL 9 support that this mitigation isn't necessary.